### ISSUE TITLE
Guidance on class definitions and training for credit statement model | 关于训练信用卡对账单模型时类别定义与训练策略的咨询

### ISSUE THREAD

**Issue by Spr-Luis** [opendatalab/DocLayout-YOLO#132]
Hi! 👋,

I'm working on a personal project to fine-tune DocLayout-YOLO for **credit card statement layout analysis**.

I’d like to ask:

1. **Recommended classes**:  
   I plan to annotate classes like `credit_summary_table`, `usage_level`, `transactions_table`, `logo_*`, etc.  
   Are there any suggested label sets for financial documents, or should I build on DocLayNet/DocStructBench classes?

2. **Training strategy**:  
   Should I train from scratch or fine-tune your pre-trained model (e.g., `doclayout_yolo_docstructbench_imgsz1024.pt`)?

---

你好团队，

我正在做一个个人项目，想基于 DocLayout-YOLO 识别**信用卡对账单**中的结构信息。

请问：

1. 是否有推荐的**标签体系**？我计划标注类似 `credit_summary_table`、`usage_level`、`transactions_table`、`logo_*` 等字段。

2. 是否建议**从头训练模型**，还是在已有模型基础上微调更合适？

---

Thanks for your great work and any advice you can share 🙏  
感谢你们的出色开源工作与指导！

**Comment by JulioZhao97** [comment]
@Spr-Luis Hello! 
1. There is no recommended label-system, I suggest you label you data closely adheres to your requirements. For example, for convenient usage in MinerU, we train DocLayout-YOLO model labeling all unnecessary document elements as ```abandon``` (such as page number, footer, and so on). Secondly, make sure the consistency of label definition, for example, if you want to detect span-level text, all text you should label is span-level.
2. I suggest you train model from [pretraining model](https://huggingface.co/juliozhao/DocLayout-YOLO-DocSynth300K-pretrain) trained on DocSynth300K, [model you mentioned](doclayout_yolo_docstructbench_imgsz1024.pt) is a fine-tuned model which is finetuned on 10K-level data. But there maybe a change document scenario can behaves differently from credit card, so you can try different training strategies and pretraining models.

If you have any further questions, feel free to let mo know!

**Comment by Spr-Luis** [comment]
> [@Spr-Luis](https://github.com/Spr-Luis) Hello!
> 
> 1. There is no recommended label-system, I suggest you label you data closely adheres to your requirements. For example, for convenient usage in MinerU, we train DocLayout-YOLO model labeling all unnecessary document elements as `abandon` (such as page number, footer, and so on). Secondly, make sure the consistency of label definition, for example, if you want to detect span-level text, all text you should label is span-level.
> 2. I suggest you train model from [pretraining model](https://huggingface.co/juliozhao/DocLayout-YOLO-DocSynth300K-pretrain) trained on DocSynth300K, [model you mentioned](doclayout_yolo_docstructbench_imgsz1024.pt) is a fine-tuned model which is finetuned on 10K-level data. But there maybe a change document scenario can behaves differently from credit card, so you can try different training strategies and pretraining models.
> 
> If you have any further questions, feel free to let mo know!

Thank you very much for the thoughtful response! I really liked the idea of inverting the approach and labeling only the *necessary* elements while marking others as `abandon` — I’ll definitely apply that.

Since you mentioned MinerU, I wanted to ask:  
Is there any way to make **custom labels** recognizable by MinerU?  
Or would I need to stick to a specific label schema for compatibility?

In my case, I’ll be mostly working with **tables**, but with highly specific structures that I’d like to distinguish — for example: `Table_Type_A`, `Table_Type_B`, etc., representing different segments or purposes.

Any advice or suggestions on how to structure this for downstream use in MinerU would be greatly appreciated!

Thanks again for your guidance 🙌

**Comment by JulioZhao97** [comment]
@Spr-Luis As far as I concern it is difficult to make custom labels in MinerU, because these labels are determined by layout models. I think what you can do is to find a way of working with MinerU labels. Unless you can train your own model and replace the layout model in MinerU. Lastly, since you mentioned table, I have a DocLayout-YOLO model trained on WTW, which can detect table cell, if this model can help you please let me know.

**Comment by Spr-Luis** [comment]
I’d love to try the WTW-trained model for table cell detection!

I think it could work great for the kind of tables I’m dealing with.
Really appreciate it!

**Comment by JulioZhao97** [comment]
@Spr-Luis WTW model is shared here: https://drive.google.com/file/d/13QhvN7jCJPHPrLhCm5hNywn43Fn6vmj5/view?usp=drive_link, and make sure the ```dfl``` is set to 0:

https://github.com/opendatalab/DocLayout-YOLO/blob/7c4be36bc61f11b67cf4a44ee47f3c41e9800a91/doclayout_yolo/nn/modules/head.py#L34

change this line to ```self.reg_max = 1```
